src/env.py

```
@@ -190,6 +172,21 @@ def step(
+        """Decode the side action into per-mallet discrete moves and apply them in order."""
+        base = len(ACTIONS)
+        if action_encoded < 0 or action_encoded >= base ** len(mallets):
+            raise ValueError(f"Invalid side action {action_encoded} for {len(mallets)} mallets")
+        # Decode into digits in base `base`
+        actions: List[int] = []
+        a = action_encoded
+        for _ in range(len(mallets)):
+            actions.append(a % base)
+            a //= base
+        # Apply actions in fixed order
+        for mallet, act in zip(mallets, actions):
```

Ruff (0.12.2)

`zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)